src/code_ef070_len_3400.c

```diff
@@ -1,10 +1,100 @@
 #include "common.h"
 
-INCLUDE_API_ASM("code_ef070_len_3400", SetCamEnabled);
-
-INCLUDE_API_ASM("code_ef070_len_3400", SetCamFlag80);
-
-INCLUDE_API_ASM("code_ef070_len_3400", SetCamPerspective);
+ApiStatus SetCamEnabled(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 id = get_variable(script, *args++);
+    s32 enabled = get_variable(script, *args++);
+
+    if (!enabled) {
+        (&gCameras[id])->flags |= 0x2;
+    } else {
+        (&gCameras[id])->flags &= ~0x2;
+    }
+    return ApiStatus_DONE2;
+}
+
+ApiStatus SetCamFlag80(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 id = get_variable(script, *args++);
+    s32 enabled = get_variable(script, *args++);
+
+    if (!enabled) {
+        (&gCameras[id])->flags |= 0x80;
+    } else {
+        (&gCameras[id])->flags &= ~0x80;
+    }
+    return ApiStatus_DONE2;
+}
+
+ApiStatus SetCamPerspective(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 id = get_variable(script, *args++);
+    s16 mode = get_variable(script, *args++);
+    f32 vfov = get_float_variable(script, *args++);
+    s16 nearClip = get_variable(script, *args++);
+    s16 farClip = get_variable(script, *args++);
+    Camera* cameras = &gCameras;
+    Camera* camera = &cameras[id];
+
+    camera->farClip = farClip;
+    camera->mode = mode;
+    camera->unk_06 = 1;
+    camera->unk_08 = 1;
+    camera->vfov = vfov;
+    camera->nearClip = nearClip;
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802CA90C(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 id = get_variable(script, *args++);
+    s16 mode = get_variable(script, *args++);
+    Camera* cameras = &gCameras;
+    Camera* camera = &cameras[id];
+
+    camera->mode = mode;
+    camera->unk_06 = 0;
+    return ApiStatus_DONE2;
+}
+
+INCLUDE_API_ASM("code_ef070_len_3400", func_802CA988);
+/*
+ApiStatus func_802CA988(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* ptrReadPos = script->ptrReadPos;
+    Bytecode id = get_variable(script, *ptrReadPos++);
+    Bytecode outVar1 = *ptrReadPos++;
+    Bytecode outVar2 = *ptrReadPos++;
+    Bytecode outVar3 = *ptrReadPos++;
+    Bytecode outVar4 = *ptrReadPos++;
+    f32 temp1, temp2, temp3;
+    f32 tempSqrt;
+
+    (&gCameras[id])->mode = 2;
+    (&gCameras[id])->unk_1C = -1 * round((&gCameras[id])->currentPitch);
+    (&gCameras[id])->unk_1E = -1 * (&gCameras[id])->currentBlendedYawNegated;
+    (&gCameras[id])->unk_06 = 0;
+
+    temp1 = (&gCameras[id])->lookAt_obj[0] - (&gCameras[id])->lookAt_eye[0];
+    temp2 = (&gCameras[id])->lookAt_obj[1] - (&gCameras[id])->lookAt_eye[1];
+    temp3 = (&gCameras[id])->lookAt_obj[2] - (&gCameras[id])->lookAt_eye[2];
+
+    tempSqrt = sqrtf((temp1 * temp1) + (temp2 * temp2) + (temp3 * temp3));
```

Comment:
you can probably remove the tempSqrt var and just put the assignment in the body of the if statement here to get this to match. Not required for the PR, but just a suggestion